lib/Model/AbstractModel.php

```diff
@@ -29,6 +30,7 @@
 abstract class AbstractModel implements ModelInterface
 {
     use ObjectVarTrait;
+    use RecursionBlockingEventDispatchHelperTrait;
```

Comment:
Since `RecursionBlockingEventDispatchHelperTrait` isn't used for all implementations that use `AbstractModel` I wouldn't include it here, but rather use it individually on the classes that make use of `dispatchEvent()`.

Reply:
What about the __sleep? can i keep it in `AbstractModel`?

```
public function __sleep()
    {
        $blockedVars = ['dao', 'o_dirtyFields', 'activeDispatchingEvents'];
```

or should i also add
```
   public function __sleep()
    {
        $parentVars = parent::__sleep();
        $blockedVars = ['activeDispatchingEvents'];

        return array_diff($parentVars, $blockedVars);
    }
```
in each class?

Reply:
the `__sleep` we can leave as it is, as it doesn't hurt 😉